src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState, forwardRef } from "react";
+import cx from "classnames";
+import moment from "moment";
+import "react-dates/lib/css/_datepicker.css";
+import { DayPickerSingleDateController, DayPickerRangeController, DayOfWeekShape } from "react-dates";
+import DatePickerHeaderComponent from "./DatePickerHeader/DatePickerHeader";
+import DateNavigationItem from "./DateNavigationItem/DateNavigationItem";
+import YearPicker from "./YearPicker/YearPicker";
+import { DAY_SIZE, WEEK_FIRST_DAY } from "./constants";
+import { Moment, FocusInput, Direction, RangeDate } from "./types";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import styles from "./DatePicker.module.scss";
+import "react-dates/initialize";
+interface DatePickerProps extends VibeComponentProps {
+  firstDayOfWeek: DayOfWeekShape;
+  date?: Moment;
+  endDate?: Moment;
+  onPickDate: (date: Moment | RangeDate) => void;
+  hideNavigationKeys: boolean;
+  enableOutsideDays: boolean;
+  showWeekNumber: boolean;
+  daySize: number;
+  shouldBlockDay?: (date: Moment) => boolean;
+  range: boolean;
+  numberOfMonths: number;
+  shouldBlockYear?: (year: number) => boolean;
+  isOutsideRange?: (date: Moment) => boolean;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const NOOP = () => {};
+
+const DatePicker: VibeComponent<DatePickerProps, HTMLElement> = forwardRef<HTMLDivElement, DatePickerProps>(
+  (
+    {
+      id,
+      className,
+      firstDayOfWeek = WEEK_FIRST_DAY,
+      daySize = DAY_SIZE,
+      range = false,
+      shouldBlockDay,
+      shouldBlockYear,
+      numberOfMonths = 1,
+      hideNavigationKeys = false,
+      date,
+      endDate,
+      onPickDate,
+      enableOutsideDays = false,
+      showWeekNumber = false,
+      isOutsideRange,
```

Comment:
we do not use is prefix or similar in component props as standard because it's less readable while declaring component instance with boolean props with shorthand syntax.
let's convert it to: outsideRange ?